### DISCUSSION TITLE
app hung after automated upgrade to 3.0.3 on azure

### DISCUSSION THREAD

**Discussion by eauio** [oqtane/oqtane.framework#2016] [General] [answered]
Running Oqtane 3.0.2 on Azure, used the admin dashboard system update option per https://www.oqtane.org/blog/!/25/upgrading-oqtane. The app is hung ( 4 hours) with Please Wait... Upgrade In Progress... (this process can take a few minutes... please be patient).

Resorted to a manual process (which I have done several times ok on IIS) by unzipping upgrade package into a azure kudu page.
Not sure if this was appropriate, my /site files now look like this .. 

<img width="799" alt="kudu-site" src="https://user-images.githubusercontent.com/6491840/155179075-89927d5b-750c-4ad8-8f61-388890401f92.PNG">

The site nevertheless continues to run the Upgrade In Progress message, any advice welcome.
Can I resolve it with a clean install, and how do I process that in Azure ?

**Comment by sbwalker** [comment]
In order to do a manual upgrade you would need to deploy the Oqtane.Framework.3.0.3.Upgrade.zip to the C:\home\site\wwwroot folder. It appears that you may have deployed it to the C:\home\site folder?

**Reply by eauio** [reply]
Thank you sbwalker, all really appreciated. Tried again with that, restarted app hung, possibly from the 1st auto update ? 
Now, successfully restored my 3.0.2 site from snapshot in azure.  The helpful admin dashboard system update option worked for me in IIS.
Want to go again with auto-update in Azure, in view of my app hang experience, I'll give it a few days to observe any feedback here.

**Comment by sbwalker** [comment, marked as answer]
The System Update option from within the framework follows these steps:

- downloads the Updater.nupkg and Framework.nupkg from Nuget.org to the /Packages folder and installs them
- downloads the upgrade ZIP file from Github.com to the /Packages folder
- creates an app_offline.htm file in the root folder (this will shut down the application - which sometimes takes 3-4 minutes so the system needs to poll to verify if any files are locked)
- creates a /Backup folder and makes a copy of all existing files
- extracts files from the ZIP
- removes /Backup folder
- removes app_offline.htm

There is obviously a lot of defensive logic to ensure it can reverse the process (ie. restore files, etc...) if any step fails.

If you run into difficulty when executing it on Azure I would suggest you check if:

- the upgrade ZIP file exists in the /Packages folder
- the app_offline.htm file exists in the root folder
- the /Backup folder exists and contains files

Currently the Updater logs messages to the Console but it would obviously be better if it saved these messages in a log file. I can add this enhancement.

**Reply by sbwalker** [reply]
@eauio #2026 creates a log file for the automated upgrade process to make this easier to troubleshoot in the future

**Reply by sbwalker** [reply]
@eauio Here is an example of the log file created after performing an automated upgrade on IIS from version 3.0.3 to 3.1.0:

Log FIle: \Packages\Oqtane.Framework.3.1.0.Upgrade.log

Upgrade Process Started: 4/12/2022 2:01:39 PM
ContentRootPath: C:\Users\Shaun.Walker\source\repos\sbwalker\Oqtane.Upgrade
WebRootPath: C:\Users\Shaun.Walker\source\repos\sbwalker\Oqtane.Upgrade\wwwroot
Located Upgrade Package: C:\Users\Shaun.Walker\source\repos\sbwalker\Oqtane.Upgrade\Packages\Oqtane.Framework.3.1.0.Upgrade.zip
Stopping Application Using: C:\Users\Shaun.Walker\source\repos\sbwalker\Oqtane.Upgrade\app_offline.htm
Retrieving List Of Files From Upgrade Package...
Preparing Backup Folder: C:\Users\Shaun.Walker\source\repos\sbwalker\Oqtane.Upgrade\Backup
Backing Up Files...
Extracting Files From Upgrade Package...
Removing Backup Folder...
Restarting Application By Removing: C:\Users\Shaun.Walker\source\repos\sbwalker\Oqtane.Upgrade\app_offline.htm
Upgrade Process Ended: 4/12/2022 2:03:09 PM
